### PR TITLE
Fixes to clear warnings from win11-23h2

### PR DESCRIPTION
**Pull request by edthedev**
## Context

To support collaboration with EPS, and potentially other uses, we need to add a configuration file to `/configs` with the configuration for Windows 11, based on the CIS Benchmarks for Windows 11.

### CODE REVIEW

**Comment by edthedev** [comment]
Adding the 'blocker' label to this as it blocks our next round of testing with the endpoint team.

**Comment by edthedev** [comment]
Current error - failing on some things that static_corrections covers:

```
PS C:\Users\delaport\src\SecOps-Powershell-CISDSC> .\tools\generate_resources_changes.ps1
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\BITLOCKER\{D2A2C776-DA56-4232-80E5-282ECA6B206F}\DomainSysvol\GPO\Machine\microsoft\windows nt\SecEdit\GptTmpl.inf
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\COMP-L1\{69A7A0B7-F996-44EF-876B-0E9811E81747}\DomainSysvol\GPO\Machine\microsoft\windows nt\SecEdit\GptTmpl.inf
InvalidOperation: C:\Users\delaport\src\SecOps-Powershell-CISDSC\src\CISDSCResourceGeneration\functions\private\ConvertFrom-SystemAccessRawGPO.ps1:43:13
Line |
  43 |              $SecuritySetting = $ValueData  
     |              ~~~~~~~~~~~~~~~~
     | A null key is not allowed in a hash literal.
Get-RecommendationFromGPOHash: C:\Users\delaport\src\SecOps-Powershell-CISDSC\src\CISDSCResourceGeneration\functions\private\ConvertFrom-SystemAccessRawGPO.ps1:47:69       
Line |
  47 |  … um = Get-RecommendationFromGPOHash -GPOHash $SystemAccessHash -Type ' …     
     |                                                ~~~~~~~~~~~~~~~~~
     | Cannot bind argument to parameter 'GPOHash' because it is null.
InvalidOperation: C:\Users\delaport\src\SecOps-Powershell-CISDSC\src\CISDSCResourceGeneration\functions\private\ConvertFrom-SystemAccessRawGPO.ps1:49:41
Line |
  49 |          $SystemAccessHash['Name'] = "'$($SystemAccessHash['Name'])'"
     |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot index into a null array.
InvalidOperation: C:\Users\delaport\src\SecOps-Powershell-CISDSC\src\CISDSCResourceGeneration\functions\private\ConvertFrom-SystemAccessRawGPO.ps1:49:41
Line |
  49 |          $SystemAccessHash['Name'] = "'$($SystemAccessHash['Name'])'"
     |                                          ~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot index into a null array.
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\COMP-L2\{B1A9318D-1F9B-48B5-8227-560CA80A2D83}\DomainSysvol\GPO\Machine\microsoft\windows nt\SecEdit\GptTmpl.inf
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\SERVICES-L1\{9D413398-293E-42C9-82AB-C8D602EB306B}\DomainSysvol\GPO\Machine\microsoft\windows nt\SecEdit\GptTmpl.inf
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\SERVICES-L2\{4747F6CC-7487-4DBE-9BF8-DC22E74B52DD}\DomainSysvol\GPO\Machine\microsoft\windows nt\SecEdit\GptTmpl.inf
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\COMP-L1\{69A7A0B7-F996-44EF-876B-0E9811E81747}\DomainSysvol\GPO\Machine\microsoft\windows nt\Audit\audit.csv
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\BITLOCKER\{D2A2C776-DA56-4232-80E5-282ECA6B206F}\DomainSysvol\GPO\Machine\registry.pol
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\FVE:FDVAllowSoftwareEncryptionFailover.
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\COMP-L1\{69A7A0B7-F996-44EF-876B-0E9811E81747}\DomainSysvol\GPO\Machine\registry.pol
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\AppHVSI:AppHVSIClipboardFileType.
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\PowerShell\ScriptBlockLogging:EnableScriptBlockInvocationLogging.
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\PowerShell\Transcription:OutputDirectory.
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows\PowerShell\Transcription:EnableInvocationHeader.
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\COMP-L2\{B1A9318D-1F9B-48B5-8227-560CA80A2D83}\DomainSysvol\GPO\Machine\registry.pol
WARNING: Failed to find a recommendation for Registry HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Windows\WCN\Registrars:MaxWCNDeviceNumber.
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\USER-L1\{F848BDFD-681B-443E-9A54-B61CD13BAB10}\DomainSysvol\GPO\User\registry.pol
VERBOSE: Importing C:\Users\delaport\src\SecOps-Powershell-CISDSC\cis_files\Windows11v2.0.0\Windows11v2.0.0\USER-L2\{67314428-4119-4B68-BE65-0840A8ED96BC}\DomainSysvol\GPO\User\registry.pol
Exception: Missing parameter validation for recommendation 1.2.4
```

**Comment by edthedev** [comment]
# TODO: Add more regression tests to try to isolate why we get a warning about AppHVSI:Etc when we generate resources, even though it is in static corrrecionts.

**Comment by edthedev** [comment]
Per today's session, this is ready for review.

**Comment by edthedev** [comment]
Note that this PR does not add the generated resources, but changes necessary to allow the Win11 resource to be generated.
Expect a follow-up pull request with the actual generated resources.

File: tools/generate_resources_example.ps1 [hunk 1]
```diff
@@ -0,0 +1,28 @@
+Import-Module '.\src\CISDSCResourceGeneration\CISDSCResourceGeneration.psd1' -Force
+Remove-Item -Path '.\output' -Recurse -Force -ErrorAction 'SilentlyContinue'
+
```
Comment (mpitcel) [inline tools/generate_resources_example.ps1:3]: Should actual NetIDs be removed from these examples?

Comment (baristaTam) [reply tools/generate_resources_example.ps1:3]: You're right. I forget this is a public repo. Please do not approve until I make those changes.

**Review by baristaTam — COMMENTED** [review]
Need to update changelog first.

File: tools/changes.txt
Comment (edthedev) [inline tools/changes.txt:1]: This file looks like it may have been added by accident?

**Review by edthedev — COMMENTED** [review]
`changes.txt` looks like it may have been added by accident?

File: tools/changes.txt
Comment (mpitcel) [reply tools/changes.txt:1]: This may be the case?
It seems like this is present for `CHANGELOG` type reasons.

Comment (edthedev) [reply tools/changes.txt:1]: @baristaTam I think this is a temporary file we user to debug something? But I'm not 100%s sure.

**Review by mpitcel — APPROVED** [review]

File: tools/changes.txt
Comment (baristaTam) [reply tools/changes.txt:1]: Yes this is an artifact from testing and should be removed.

**Comment by baristaTam** [comment]
'CIS_Microsoft_Windows_11_Enterprise_Release_23H2' needs to be added to the DscResourcesToExport dictionary in CISDSC.psd1

**Review by mpitcel — APPROVED** [review]

**Review by baristaTam — APPROVED** [review]

**Pull request merged**